cmake/CMakeLists.txt

```
@@ -691,11 +691,8 @@ if(BUILD_SHARED_LIBS OR PKG_PYTHON)
     find_package(Python COMPONENTS Interpreter)
   endif()
   if (Python_EXECUTABLE)
-    execute_process(COMMAND ${Python_EXECUTABLE}
-      -c "import distutils.sysconfig as cg; print(cg.get_python_lib(1,0,prefix='${CMAKE_INSTALL_PREFIX}'))"
-      OUTPUT_VARIABLE PYTHON_DEFAULT_INSTDIR OUTPUT_STRIP_TRAILING_WHITESPACE)
-    set(PYTHON_INSTDIR ${PYTHON_DEFAULT_INSTDIR} CACHE PATH "Installation folder for LAMMPS Python module")
-    install(FILES ${LAMMPS_PYTHON_DIR}/lammps.py DESTINATION ${PYTHON_INSTDIR})
+    file(MAKE_DIRECTORY ${CMAKE_BINARY_DIR}/python)
+    install(CODE "execute_process(COMMAND ${Python_EXECUTABLE} setup.py build -b ${CMAKE_BINARY_DIR}/python install --prefix=${CMAKE_INSTALL_PREFIX} WORKING_DIRECTORY ${LAMMPS_PYTHON_DIR})")
```

Most packagers rely on the injection of `DESTDIR` to change install path (but leaving rpaths and all the other magic the same), so I am wondering if this mechanism will work here.

---

Ok, we have a problem here:
```
cmake ../cmake/ -DPKG_PYTHON=ON -DCMAKE_INSTALL_PREFIX=/usr
make -j3
make install DESTDIR=$PWD/test
```
fails with:
```
creating /usr/lib/python3.9/site-packages/lammps
error: could not create '/usr/lib/python3.9/site-packages/lammps': Permission denied
```

So `DESTDIR` is missing.

---

@junghans Can you show me one of those injections?

---

```suggestion
    install(CODE "execute_process(COMMAND ${Python_EXECUTABLE} setup.py build -b ${CMAKE_BINARY_DIR}/python install --prefix=${CMAKE_INSTALL_PREFIX} --root $DESTDIR WORKING_DIRECTORY ${LAMMPS_PYTHON_DIR})")
```
works for make at least.

---

Works with `DESTDIR=$PWD/test ninja install` as well.

---

```suggestion
    install(CODE "execute_process(COMMAND ${Python_EXECUTABLE} setup.py build -b ${CMAKE_BINARY_DIR}/python install --prefix=${CMAKE_INSTALL_PREFIX} --root \$ENV{DESTDIR}/ WORKING_DIRECTORY ${LAMMPS_PYTHON_DIR})")
```
seems to work in all case, w/o DESTDIR and make/ninja.